gdtoolkit: init at 3.2.8

critbase opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change
To package https://github.com/Scony/godot-gdscript-toolkit, an "independent set of GDScript tools."

Had this branch laying around for a while, forgot to upstream it until now.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

critbase committed:
gdtoolkit: init at 3.2.8

risicle commented:
You haven't referenced it from the top-level anywhere (e.g. `all-packages.nix`) so as it is this package is unusable.

risicle commented on pkgs/development/tools/gdtoolkit/default.nix:
```diff
@@ -0,0 +1,31 @@
+    description =
+      "Independent set of GDScript tools - parser, linter and formatter";
+    license = licenses.mit;
+    platforms = platforms.linux;
```
Seems to work fine for me on darwin too so you can probably safely change this to `platforms.unix`.

critbase committed:
add critbase to maintainer-list.nix

critbase commented:
Argh, I knew I forgot something. Fixed, thank you. This time I tested to make sure `nix-build -A gdtoolkit` works.
I also added myself to the maintainer-list, because I apparently forgot to.

risicle reviewed (approved):
Builds & runs, macos 10.15 & nixos x86_64

SuperSandro2000 commented on pkgs/development/tools/gdtoolkit/default.nix:
```diff
@@ -0,0 +1,31 @@
+
+  meta = with lib; {
+    homepage = "https://github.com/Scony/godot-gdscript-toolkit";
+    description =
+      "Independent set of GDScript tools - parser, linter and formatter";
```
```suggestion
    description = "Independent set of GDScript tools - parser, linter and formatter";
```

SuperSandro2000 commented on pkgs/development/tools/gdtoolkit/default.nix:
```diff
@@ -0,0 +1,31 @@
+  buildInputs = [ lark-parser docopt pyyaml setuptools ];
+  propagatedBuildInputs = [ lark-parser docopt pyyaml setuptools ];
+
+  preConfigure = ''
+    sed -i 's/lark-parser==0.8.0/lark-parser>=0.8.0/' setup.py
+  '';
```
```suggestion
  postPatch = ''
    substituteInPlace setup.py \
      --replace "lark-parser==0.8.0" "lark-parser>=0.8.0"
  '';
```

SuperSandro2000 commented on pkgs/development/tools/gdtoolkit/default.nix:
```diff
@@ -0,0 +1,31 @@
+    sha256 = "3+zr1JcCt8HPwDG0j+ttnzGOhvUmaz+qzbE2wGyT86M=";
+  };
+
+  buildInputs = [ lark-parser docopt pyyaml setuptools ];
+  propagatedBuildInputs = [ lark-parser docopt pyyaml setuptools ];
```
```suggestion
  propagatedBuildInputs = [ lark-parser docopt pyyaml setuptools ];
```

critbase commented:
Just found out why this package required lark-parser 0.8.0... just pushed a version that overrides the version of lark-parser used, and made sure to test the binaries on actual `.gd` files.. I should really test my own PRs a lot more thoroughly.
Please review this version.

SuperSandro2000 commented on pkgs/development/tools/gdtoolkit/default.nix:
```diff
@@ -0,0 +1,35 @@
+
+  propagatedBuildInputs = [ lark-parser080 docopt pyyaml setuptools ];
+
+  doCheck = false;
```
Please comment why tests are disabled? Can we add a pythonImportsCheck instead?

SuperSandro2000 commented on pkgs/development/tools/gdtoolkit/default.nix:
```diff
@@ -0,0 +1,35 @@
+    homepage = "https://github.com/Scony/godot-gdscript-toolkit";
+    description = "Independent set of GDScript tools - parser, linter and formatter";
+    license = licenses.mit;
+    platforms = platforms.unix;
```
```suggestion
```
set by buildPythonApplication

hesiod commented:
This PR can be closed, gdtoolkit was added in #160040
